algolia.js

```diff
@@ -93,11 +88,25 @@ async function index() {
 
   const raw = fs.readFileSync('dist/pages.json', 'utf-8');
   /** @type {AlgoliaItem[]} */
-  const algoliaData = JSON.parse(raw).map((e) => {
-    // Set date of when object is being added to algolia.
-    e.indexedOn = indexedOn.getTime();
-    return e;
-  });
+  const algoliaData = JSON.parse(raw)
+    .map((/** @type {AlgoliaItem} */ e) => {
+      addLocaleToUrl(e.locale, e.url);
+      // Set date of when object is being added to algolia.
+      e.indexedOn = indexedOn.getTime();
+      return e;
+    })
+    .map((/** @type {AlgoliaItem} */ e) => {
+      if (e.locale === defaultLocale) {
+        const locales = urlToLocale[e.url];
+        e.locales = [
+          e.locale,
+          ...supportedLocales.filter((i) => locales.indexOf(i) === -1),
```

Comment:
Use isSupportedLocale instead. You can also do this check when you build urlToLocales map instead of post-processing here.

Reply:
So this filter removes the locales that do exist as urls from the `supportedLocales` array. Where as `isSupportedLocale` ensures that the locale does exist. They kind of do the opposite thing. So I don't believe I can do what you're suggesting.